src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
```

Comment:
PrimathonAICodeReviewer

Consider adding a comment above the class declaration to describe the purpose of `ResponseInterceptor`.

```typescript
// Interceptor to format API responses
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->